src/context/shopping-cart.context.tsx

```diff
@@ -0,0 +1,39 @@
+import { createContext, ReactNode, useState } from "react";
+import { CartItem } from "../interfaces/cart/cart.interface";
+
+interface ShoppingCartContextType {
+  shoppingCartItems: CartItem[],
+  addItemToCart: (item: CartItem) => void,
+  deleteItemFromCart: (id: string) => void,
+}
+
+interface ShoppingCartProviderProps {
+  children: ReactNode;
+}
+
+export const ShoppingCartContext = createContext<ShoppingCartContextType>({} as ShoppingCartContextType);
+
+export function ShoppingCartProvider({ children }: ShoppingCartProviderProps) {
+  const [shoppingCartItems, setShoppingCartItems] = useState<CartItem[]>([]);
+
+  const addItemToCart = (cartItem: CartItem) => {
+    const itemExistsInTheCart: boolean = shoppingCartItems.find(item => cartItem.id === item.id) !== undefined;
+
+    if (itemExistsInTheCart) {
+      setShoppingCartItems((prevItems) => prevItems.map((item) => item.id === cartItem.id ? {...item, quantity: item.quantity + 1} : item))
```

Comment:
Not a big fan of this approach with prevItems in this case.
Basically we should find the item in the cart. If it exists then we update it with `setShoppingCartItems` with new quantity else we set the quantity to 1.
````
const cartItem= cart.find(...);
if (cartItem) {
  if (cartItem.quantity > 1) {
    cartItem.quantity += 1;
  } else {
    cart.push({
        ...cartItem,
        quantity: 1,
      });
}
````
Something like this. My code needs changes. It's just for guidance.